es-ds-components/components/es-modal.vue

```diff
@@ -75,9 +79,12 @@ const getSizeClass = computed(() => {
         modal
         :class="getSizeClass"
         :pt="modalPt"
+        :closable="!hideHeader"
```

Comment:
I'd recommend controlling this separately from the header, by exposing a new boolean `closeable` prop, defaulted to true, and passing it in here.